packages/react/__tests__/__snapshots__/server.js.snap

```diff
@@ -7,7 +7,7 @@ exports[`ssr @import 1`] = `
 <div class="css-1lrxbo5">
 </div>
 <style data-emotion="css-global 1sh9nmh">
-  @import url('https://some-url');h1{color:hotpink;}
+  h1{color:hotpink;}@import url('https://some-url');
```

Comment:
`@import` has to be the first rule(or at least before non `@import` rules) when using insertRule, could you check if this not being first here is fine?

Reply:
It probably is not - I've changed tests to always put `@import` first: https://github.com/emotion-js/emotion/pull/1817/commits/2d95e2c3c349e0d2c1d352f0c0fb2fd8fd02c2ee

Stylis v3 was special-casing this and it actually has moved import rules to the first position when compiling, v4 doesn't do that anymore. It seems to me that it's a valid approach to just require from the user to put those first "manually". Special-casing this would require from us writing a special stylis plugin for the server path - IMHO it's not worth it. 

If we decide this being a requirement on the consumer side then this special case could be dropped I think: https://github.com/emotion-js/emotion/blob/ff9746b8d6f5e8dca89bc9a7340a90b91240800b/packages/sheet/src/index.js#L118
`@import` rules should not occur in the scoped styles and for global ones they should just be put first. Keep also in mind that `@charset` should have even higher priority than `@import` and we do not special case it anyway - it's way less common though.

Reply:
> Keep also in mind that `@charset` should have even higher priority than `@import` and we do not special case it anyway - it's way less common though.

Yeah, I'm not super worried about `@charset` given I've never seen anyone use it.

Okay - I think I'm fine with requiring `@import` be first and only be in `Global`. We definitely need a better error than what we would have if we just removed the isImportRule stuff though.

Reply:
Ok, I've pushed out changes around this - together with new dev-only warnings for incorrect inputs.